### ISSUE TITLE
Launch the Python interpreter using the `-I` flag

### ISSUE THREAD

**Issue by carlosperate** [mu-editor/pup#195]
To isolate the Python packages from the user environment.

Basically we have some cases where the user environment can have other versions of the packages Mu uses, which then take precedence, and cause issues.

In this example the user's PyQt5 package is different, it gets picked up by Mu and then throws an exception:
- https://github.com/mu-editor/mu/issues/1602#issuecomment-1000907359

```
Traceback (most recent call last):                                                                                      
  File "C:\Users\Bens Pc\AppData\Local\Programs\Mu Editor\Python\lib\runpy.py", line 194, in _run_module_as_main        
    return _run_code(code, main_globals, None,                                                                          
  File "C:\Users\Bens Pc\AppData\Local\Programs\Mu Editor\Python\lib\runpy.py", line 87, in _run_code                   
    exec(code, run_globals)                                                                                             
  File "C:\Users\Bens Pc\AppData\Local\Programs\Mu Editor\Python\lib\site-packages\mu\__main__.py", line 1, in <module> 
    from mu.app import run                                                                                              
  File "C:\Users\Bens Pc\AppData\Local\Programs\Mu Editor\Python\lib\site-packages\mu\app.py", line 46, in <module>     
    from .interface import Window                                                                                       
  File "C:\Users\Bens Pc\AppData\Local\Programs\Mu Editor\Python\lib\site-packages\mu\interface\__init__.py", line 1, in <module>                                                                                                               
    from .main import Window                                                                                            
  File "C:\Users\Bens Pc\AppData\Local\Programs\Mu Editor\Python\lib\site-packages\mu\interface\main.py", line 65, in <module>                                                                                                               
    from mu.interface.editor import EditorPane                                                                          
  File "C:\Users\Bens Pc\AppData\Local\Programs\Mu Editor\Python\lib\site-packages\mu\interface\editor.py", line 25, in <module>                                                                                                               
    from PyQt5.Qsci import (                                                                                            
ImportError: DLL load failed while importing Qsci: The specified procedure could not be found.
```

Checking the packages origin using pip:

```
C:\Users\Bens Pc>"%LOCALAPPDATA%\Programs\Mu Editor\Python\python.exe" -m pip list --verbose
Package                           Version      Location                                                                                               Installer
--------------------------------- ------------ ------------------------------------------------------------------------------------------------------ ---------
...
mu-editor                         1.1.0b7      c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages
...
PyQt5                             5.15.2       c:\users\bens pc\appdata\roaming\python\python38\site-packages                                         pip
pyqt5-plugins                     5.15.2.2.0.1 c:\users\bens pc\appdata\roaming\python\python38\site-packages                                         pip
...
pyqt5-tools                       5.15.2.3     c:\users\bens pc\appdata\roaming\python\python38\site-packages                                         pip
...
python-dotenv                     0.15.0       c:\users\bens pc\appdata\roaming\python\python38\site-packages                                         pip
...
qt5-applications                  5.15.2.2.1   c:\users\bens pc\appdata\roaming\python\python38\site-packages                                         pip
qt5-tools                         5.15.2.1.0.1 c:\users\bens pc\appdata\roaming\python\python38\site-packages                                         pip
...

```

And then using with he `-I` flag:

```
C:\Users\Bens Pc>"%LOCALAPPDATA%\Programs\Mu Editor\Python\python.exe" -I -m pip list --verbose
Package                           Version Location                                                                                               Installer
--------------------------------- ------- ------------------------------------------------------------------------------------------------------ ---------
adafruit-board-toolkit            1.1.0   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
appdirs                           1.4.4   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
backcall                          0.2.0   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
backports.entry-points-selectable 1.1.1   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
black                             21.12b0 c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
click                             8.0.3   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
colorama                          0.4.4   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
decorator                         5.1.0   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
distlib                           0.3.4   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
filelock                          3.4.0   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
flake8                            4.0.1   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
ipykernel                         5.5.6   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
ipython                           7.30.1  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
ipython-genutils                  0.2.0   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
jedi                              0.18.1  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
jupyter-client                    6.1.12  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
jupyter-core                      4.9.1   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
matplotlib-inline                 0.1.3   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
mccabe                            0.6.1   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
mu-editor                         1.1.0b7 c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages
mypy-extensions                   0.4.3   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
nudatus                           0.0.5   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
parso                             0.8.3   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
pathspec                          0.9.0   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
pickleshare                       0.7.5   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
pip                               21.1.3  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages\pip-21.1.3-py3.8.egg
platformdirs                      2.4.0   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
prompt-toolkit                    3.0.24  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
pycodestyle                       2.8.0   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
pyflakes                          2.4.0   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
Pygments                          2.10.0  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
PyQt5                             5.13.2  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
PyQt5-sip                         12.9.0  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
PyQtChart                         5.13.1  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
pyserial                          3.5     c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
python-dateutil                   2.8.2   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
pywin32                           303     c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
pyzmq                             22.3.0  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
QScintilla                        2.11.3  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
qtconsole                         4.7.7   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
QtPy                              1.11.3  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
semver                            2.13.0  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
setuptools                        57.4.0  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages\setuptools-57.4.0-py3.8.egg
six                               1.16.0  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
tomli                             1.2.3   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
tornado                           6.1     c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
traitlets                         5.1.1   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
typing-extensions                 4.0.1   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
virtualenv                        20.10.0 c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
wcwidth                           0.2.5   c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
wheel                             0.37.0  c:\users\bens pc\appdata\local\programs\mu editor\python\lib\site-packages                             pip
```

And launching Mu with the `-I` flag also resolves the PyQt5 exception.

**Comment by tmontes** [comment]
THOUGHTS:

* Could this be the issue described in #42? Feels like the same.
* From 3.6-3.10 CLI help:
```
$ python3.x --help | grep -e "^-I" -e "^-E" -e "^-s"
-E     : ignore PYTHON* environment variables (such as PYTHONPATH)
-I     : isolate Python from the user's environment (implies -E and -s)
-s     : don't add user site directory to sys.path; also PYTHONNOUSERSITE
```

LET's:

* Have `pup` package applications such that they're launched with the `-I` flag.
* Support package time overriding of that via the `--launch-py-flag` CLI option.

**Comment by tmontes** [comment]
PROGRESS:

* Just merged PR #197:
  * Packaged applications are by default launched with `-I`.
  * Can be overridden with the `--launch-pyflag` package option.

NEXT:

* Try packaging Mu and observe behaviour.
* Update docs.

**Comment by tmontes** [comment]
CLOSED after having tested Mu and updated README/docs.
